### ISSUE TITLE
[Suggestion] Turret secondary axis on grand-child entities

### ISSUE THREAD

**Issue by Kaiser-Indrasil** [tsunamayo/Starship-EVO#2697]
Aside from the notorious bug where turret's secondary axis doesn't get a checkmark until you link a weapon, I have a major beef with how prohibitive turret designs currently are. Let me explain.

Right now, turret's secondary axis/axes have to be placed on the entity produced by the rotor / hinge which handles the primary axis. This forbids any and all designs which introduce child entities in the hierarchy between the primary and secondary axes. Much like this:

![image](https://user-images.githubusercontent.com/63756057/88107791-a6fc1b80-cb9f-11ea-9474-85e2c5de2928.png)

This isn't limited to this design exclusively. Any turrets that would have extra deployable bits, would be impossible to control.
There are dozens of turret replicas which aren't feasible for me right now because of this issue.

I honestly think this is a huge drawback in a game that focuses on creative freedom, after all.

**Comment by ProPeach** [comment]
This is pretty important, especially for turrets which collapse as Indrasil's does.

Another mechanism that is critical to this kind of thing is the ability to link multiple logic blocks to one rotor/rail etc, as pointed out in many places but most notably #2617. This is really important Tsuna, it even breaks things like linking screens to rotors and other mechanisms which is another feature that I assume will be expanded later.

**Comment by tsunamayo** [comment]
Yes #2617 is on my to-do list. Can I enquire how Starmade and SE are working with turret? Can you make such complicated design?
The huge problem is collision check, I have to bake a 2D map to not run expensive collision check each time you use your turret. With more than 2 axis I cant.

**Comment by Kaiser-Indrasil** [comment]
Well, the custom turret designs in SE work thanks to player-made C# scripts which are put in Programmable Blocks:

https://steamcommunity.com/workshop/filedetails/?id=944915802

I guess the player that wrote #2652 must be very fond of them. However, since Starship EVO uses proprietary logic gate approach to in-game programming, so I'm not a huge fan of the suggestion. After all, there's people like Talrey who are making whole computers using the logic system already!

Getting back on topic, there are fully pre-fab turrets in SE and custom ones which cater to simple 2-axes approach, as well as multi-rotor-grid contraption things because the engine does not pre-bake anything. Yes, it's very unoptimised but allows for very complicated designs like this:
![image](https://user-images.githubusercontent.com/63756057/88275553-77e7c600-ccd5-11ea-862b-1785f91d8899.png)

**Comment by tsunamayo** [comment]
Thanks for the answer! How is the performance? People like to put turrets by the dozens...

**Comment by Kaiser-Indrasil** [comment]
SE is infamous for its poor performance in general but I didn't notice any performance drops directly caused by turret operation.
